File: src/main/java/com/example/chatService/controller/GroupRoomController.java
package com.example.chatService.controller;

import com.example.chatService.dto.ChatMessageResponse;
import com.example.chatService.dto.RoomResponse;
import com.example.chatService.entity.GroupMessageEntity;
import com.example.chatService.repository.ChatRoomV2Repository;
import com.example.chatService.repository.GroupMessageRepository;
import com.example.chatService.service.GroupRoomService;
import lombok.RequiredArgsConstructor;
import org.springframework.data.domain.PageRequest;
import org.springframework.web.bind.annotation.*;

import java.util.Collections;
import java.util.List;

@RestController
@RequiredArgsConstructor
@RequestMapping("/api/rooms")
public class GroupRoomController {

    private final GroupRoomService groupRoomService;

    @GetMapping("/{roomId}/messages")
    public List<ChatMessageResponse> messages(
            @PathVariable String roomId,
            @RequestParam(defaultValue = "50") int limit
    ) {
        return groupRoomService.messages(roomId, limit);
    }

    @GetMapping("/with-count")
    public List<RoomResponse> getRoomsWithCount() {
        return groupRoomService.getRoomsWithCount();
    }
}

File: src/main/java/com/example/chatService/controller/GroupRoomController.kt
package com.example.chatService.controller

import com.example.chatService.dto.ChatMessageResponse
import com.example.chatService.dto.RoomResponse
import com.example.chatService.service.GroupRoomService
import org.springframework.web.bind.annotation.GetMapping
import org.springframework.web.bind.annotation.PathVariable
import org.springframework.web.bind.annotation.RequestMapping
import org.springframework.web.bind.annotation.RequestParam
import org.springframework.web.bind.annotation.RestController

@RestController
@RequestMapping("/api/rooms")
class GroupRoomController(
    private val groupRoomService: GroupRoomService,
) {
    @GetMapping("/{roomId}/messages")
    fun messages(@PathVariable roomId: String, @RequestParam(defaultValue = "50") limit: Int): List<ChatMessageResponse> =
        groupRoomService.messages(roomId, limit)

    @GetMapping("/with-count")
    fun getRoomsWithCount(): List<RoomResponse> = groupRoomService.getRoomsWithCount()
}
